src/utill.rs

```diff
@@ -185,3 +185,58 @@ pub fn to_hex(bytes: &Vec<u8>) -> String {
 
     hex_string
 }
+
+#[cfg(test)]
+mod tests {
+    use bitcoin::blockdata::{opcodes::all, script::Builder};
+    use serde_json::json;
+
+    use super::*;
+
+    #[test]
+    fn test_convert_json_rpc_bitcoin_to_satoshis() {
+        // Test with an integer value
+        let amount = json!(1);
+        assert_eq!(convert_json_rpc_bitcoin_to_satoshis(&amount), 100_000_000);
+
+        // Test with a very large value
+        let amount = json!(12345678.12345678);
+        assert_eq!(
+            convert_json_rpc_bitcoin_to_satoshis(&amount),
+            1_234_567_812_345_678
+        );
+    }
+
+    #[test]
+    fn test_to_hex_empty_bytes() {
+        let bytes: Vec<u8> = Vec::new();
+        assert_eq!(to_hex(&bytes), "");
+    }
+
+    #[test]
+    fn test_to_hex_single_byte() {
+        let bytes: Vec<u8> = vec![0xAB];
+        assert_eq!(to_hex(&bytes), "ab");
+    }
+
+    #[test]
+    fn test_to_hex_multiple_bytes() {
+        let bytes: Vec<u8> = vec![0x12, 0x34, 0x56, 0xFF];
+        assert_eq!(to_hex(&bytes), "123456ff");
+    }
+
+    #[test]
+    fn test_redeemscript_to_scriptpubkey() {
+        // Create a custom puzzle script
+        let puzzle_script = Builder::new()
+            .push_opcode(all::OP_ADD)
+            .push_opcode(all::OP_PUSHNUM_2)
+            .push_opcode(all::OP_EQUAL)
+            .into_script();
```

Comment:
Good start. How about trying creating some more scripts like `P2PKH`, `P2SH`, `P2WSH`, and some custom script_pubkey with check multisigs and stuffs.

Check rust-bitcoin for test vectors and ideas.

Reply:
I created more scripts to test redeemscript_to_scriptpubkey function